File: my-app/src/app/models/ImageProduct.ts
export interface ImageProduct {
    id:        number;
    productId: number;
    product:   Product;
    image:     string;
    image2:    string;
    image3:    string;
    image4:    string;
}

export interface Product {
    id:              number;
    name:            string;
    imageUrl:        string;
    description:     string;
    price:           number;
    quantityInStock: number;
    categoryId:      number;
    category:        Category;
}

export interface Category {
    id:    number;
    brand: string;
    type:  string;
}

File: API/Entities/ImageProduct.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entities
{
    public class ImageProduct
    {
        public int Id {get;set;}
        public int ProductId {get;set;}
        public Product Product {get;set;}

        public string image {get;set;}
        public string image2 {get;set;}

        public string image3 {get;set;}

        public string image4 {get;set;}

    }
}